### PR TITLE
Add opam exec --no-switch

### LINKED ISSUE #4951: Have a way to temporarily remove opam from the environment (opam exec --empty)
Having `opam exec --empty` and ~`opam env --empty`~ or similiar command available would be really nice for setups where you need to take things added by opam out of the equation.

Currently I use a hack instead:
```
$ opam switch create empty --empty
$ eval $(opam env)
```
but this isn’t pretty, nor is it easy to use in scripts

EDIT: I was unaware of `opam env --revert`, but a similar thing with `opam exec` would still be nice

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Fixes #4951

### CODE REVIEW

**Comment by dra27** [comment]
It's a nice feature! Can I slightly bike-shed the name of the option, though. `opam exec` _always_ reverts the environment - it's just that its default behaviour then _sets_ the environment of the switch. It's this latter behaviour wants suppressing. Might `--no-env` therefore be slightly more appropriate (in ocaml-env, included in opam-repository-mingw, has `ocaml-env exec --no-opam` for this purpose). Using a `--no-` style option also makes it natural to include the pair of options to have the default being explicit (something we've always found to be useful in `ocamlc`/`ocamlopt` - if you add an option to turn something off, it's actually worth having the explicit option to turn it on from day 1)

**Comment by rjbou** [comment]
I've pushed an update, to revert only opam env variable. 
I've also added a test, there is a non consistent behavior between `opam env --revert` and `opam exec --revert-env` on defined variables, see [`NV_VARS`](https://github.com/ocaml/opam/pull/4957/files#diff-e61d050e60e9082edaaef66231c97c260221ce10bd518e97512a1e60de3b66a7R52)

File: src/client/opamConfigCommand.ml [hunk 1]
```diff
@@ -231,7 +231,7 @@ let expand gt str =
     (OpamFilter.expand_string ~default:(fun _ -> "")
        (OpamPackageVar.resolve st) str)
 
-let exec gt ~set_opamroot ~set_opamswitch ~inplace_path command =
+let exec gt ~set_opamroot ~set_opamswitch ~inplace_path ~revert_env command =
```
[the remark this answers is not in the document]
Comment (kit-ty-kate) [reply src/client/opamConfigCommand.ml:234]: Unless this is a user-facing function, having optional argument increases the complexity and implicit behaviours of functions. I think using them outside of user-facing functions ultimately results in more bugs.

**Comment by kit-ty-kate** [comment]
> It's a nice feature! Can I slightly bike-shed the name of the option, though. `opam exec` _always_ reverts the environment - it's just that its default behaviour then _sets_ the environment of the switch. It's this latter behaviour wants suppressing. Might `--no-env` therefore be slightly more appropriate (in ocaml-env, included in opam-repository-mingw, has `ocaml-env exec --no-opam` for this purpose). Using a `--no-` style option also makes it natural to include the pair of options to have the default being explicit (something we've always found to be useful in `ocamlc`/`ocamlopt` - if you add an option to turn something off, it's actually worth having the explicit option to turn it on from day 1)

I’m not really convinced by `--no-env`. I’m not sure what is the best naming here.

**Comment by dra27** [comment]
From dev meeting: suggestion instead to have a synergy between `opam exec --switch=<switch>` which selects the switch you want to execute in and therefore `opam exec --no-switch` to mean this behaviour (i.e. execute without a switch selected) cf. #4850

**Review by rjbou — APPROVED** [review]

**Pull request merged**